datasetinsights/commands/download.py

```diff
@@ -0,0 +1,106 @@
+import logging
+
+import click
+
+import datasetinsights.constants as const
+
+logger = logging.getLogger(__name__)
+
+
+def validate_source_uri(ctx, param, value):
```

Comment:
This could be replaced with the extension of a click.ParamType class.
example

`class SourceURI(click.ParamType):

    name = "uri"

    def convert(self, value, param, ctx):
        if os.path.exists(value) or value.startswith("gs://"):
            return value
        else:
            self.fail(f"The uri {value} has to exist on the local filesystem or be a uri to gcs ",
                      param,
                      ctx)
`

Reply:
Good call. I will incorporate this suggestion.